src/transports/dash/manifest_parser.ts

```diff
@@ -155,39 +155,74 @@ export default function generateManifestParser(
                       responseType: "arraybuffer" as const,
                       cancelSignal });
           return req;
+        }).then((res) => {
+          return {
+            ...res,
+            responseData: {
+              success: true as const,
+              data: res.responseData },
+          };
+        }, (err) => {
+          const error = formatError(err, {
+            defaultCode: "PIPELINE_PARSE_ERROR",
+            defaultReason: "An unknown error occured when parsing ressources.",
+          });
+          return {
+            size: undefined,
+            requestDuration: undefined,
+            responseData: {
+              success: false as const,
+              error },
+          };
         });
       });
 
       return PPromise.all(externalResources).then(loadedResources => {
         if (value.format === "string") {
           const resources = loadedResources.map(resource => {
-            if (typeof resource.responseData !== "string") {
-              throw new Error("External DASH resources should have been a string");
+            if (!resource.responseData.success) {
```

Comment:
Why is this code still necessary?
Couldn't we also move the `if` higher?

Reply:
Because the shape of the object is not the same when it encounters an error or when he succeed

Reply:
I'm talking about this whole block of code. Why is it not with the one L158?